From bf2071fad910dde684b0d99a23223b938eb0d559 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 20 Apr 2020 00:24:58 +0300
Subject: [PATCH] [SW-0] boot_if: rename and comment cmd clr func

The cpu_boot_cmd_clr() function's use must be reduced to only the cases
when it is absolutely necessary as the cmd/message register should be
written by LKD solely.
The only one case when this register should written by ARM is when LKD
performs the hard reset, as the reasons for performing hard reset might
render PCI not to be reliable and thus LKD clear of the register might
not propagate.
Change the function name to reflect the purpose and add a comment
describing the use.

Change-Id: I1dd7697421b9a736686f3c5178ce9e60e1763472
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
index 0a5e69f..6d848ace 100644
--- a/common/boot_if.c
+++ b/common/boot_if.c
@@ -11,7 +11,16 @@
 #include <hl_boot_if.h>
 #include <preboot.h>
 
-void cpu_boot_cmd_clr(void)
+/*
+ * The cpu_boot_cmd_clr_before_wfe() function should NOT be used in general as
+ * the LKD message register should be written by LKD solely.
+ *
+ * There is only one special case when this function can be used is only when
+ * the LKD performs hard reset, which may mean the H/W is not reliable and
+ * writing to PCI might not complete and result in the register not cleared.
+ * So to be on the safe side, when receiving the WFE, clear the message register
+ */
+void cpu_boot_cmd_clr_before_wfe(void)
 {
 	regmap_writel(KMD_MSG_NA, mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
 }
diff --git a/common/boot_if.h b/common/boot_if.h
index 359fadd..e0f54c0 100644
--- a/common/boot_if.h
+++ b/common/boot_if.h
@@ -10,7 +10,7 @@
 #include <hl_boot_if.h>
 
 enum kmd_msg cpu_boot_cmd_get(void);
-void cpu_boot_cmd_clr(void);
+void cpu_boot_cmd_clr_before_wfe(void);
 void cpu_boot_sts_set(u32 val);
 void cpu_boot_err_clrset(u32 mask, bool set);
 void cpu_boot_rsp_set(enum cpu_msg_status sts);
